srml/support/procedural/src/storage/impls.rs

```diff
@@ -283,6 +301,23 @@ impl<'a, I: Iterator<Item=syn::Meta>> Impls<'a, I> {
 			impl<#impl_trait> #scrate::storage::hashed::generator::AppendableStorageMap<#kty, #typ>
 				for #name<#trait_and_instance> #where_clause
 			{}
+
+			#visibility struct #default_delegator_ident<#struct_trait>(
+				#scrate::rstd::marker::PhantomData<(#trait_and_instance)>
+			) #where_clause;
+			impl<#impl_trait> #scrate::traits::StorageDefault<#typ>
+				for #default_delegator_ident<#trait_and_instance> #where_clause
+			{
+				fn default() -> Option<#typ> {
+					#default_delegator_return
+				}
+			}
+
+			impl<#impl_trait> #scrate::storage::hashed::generator::DecodeLengthStorageMap<#kty, #typ>
+				for #name<#trait_and_instance> #where_clause
+			{
+				type Default = #default_delegator_ident<#trait_and_instance>;
```

Comment:
(as a note: instead of having an associated type you could make it a function and get rid of StorageDefault trait.)